Configurable burst_limit

loafoe opened the pull request:
### Description

Add a ` burst_limit`  argument to the provider block to configure the helm client-go connection. The default value is `100` but should be increased in case the cluster you apply to has many CRDs. See #1011 for a more detailed explanation. The CL is large because it includes the required dependabot merge from #1016 

### Acceptance tests
- [ ] Have you added an acceptance test for the functionality being added?


### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-helm/blob/main/CHANGELOG.md):

```release-note
* Adds the burst_limit argument to configure the Helm client-go burst configuration. Useful for Crossplane enabled clusters
```
### References

Issue describing the problem in detail: #1011  
Related helm issue: https://github.com/helm/helm/issues/10560

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

marcofranssen reviewed (approved):
LGTM :rocket: 

We could use this, as we are at a stage to step away completely from this provider due to all the timeout issues we are facing.

Looking forward to see this in a release.

loafoe committed:
Implement burst_limit parameter #1011

Adds the ability to configure the burst_limit of the client-go client
used by Helm. If your cluster has a lot of CRDs e.g. when Crossplane
is installed, the DiscoveryClient performs many API calls potentially
triggering client-side rate limiting.

Increasing the burst_limit reduces this throttling, significantly
speeding up install or update actions, sometimes by a factor 10.

Co-authored-by: Marco Franssen <marco.franssen@philips.com>

BBBmau reviewed (approved):
All looks great! Thank you for contributing!
